### ISSUE TITLE
segmentation fault when executing urllib3 request from inside a PyQt5 QApplication event loop

### ISSUE THREAD

**Issue by mjb-v9-5-2** [urllib3/urllib3#1563]
PyQt5 is causing a segmention fault when a urllib3 request to read a webpage is made from inside a QApplication event loop.

The error occurs if, for example, a PyQt5 button-click is used to execute the urllib3 request.

The same error occurs if the urllib3 request is run inside the loop without any PyQt5 form being opened.

The urllib3 request executes successfully if it is run before entering the PyQt5 event loop, and is then executed again inside the event loop.

```
#! /usr/bin/env python3
from PyQt5.QtWidgets import QDialog, QApplication
import urllib3
import certifi

u = "https://www.w3.org/2019/03/pressrelease-webauthn-rec.html.en"

def test():
    http = urllib3.PoolManager( cert_reqs='CERT_REQUIRED',ca_certs=certifi.where() )
    l = http.request( 'GET', u )
    print("page status: {0}".format(l.status))
    print("page sample: {0}".format(l.data[:42]))

if input('run pre-test?').lower() == 'y': test()

if __name__=="__main__":
    A = QApplication(sys.argv)
    test()
    sys.exit( A.exec_() )
```

Environment:
```

$ conda --version
conda 4.6.11

$ conda list | grep pyqt
pyqt                      5.9.2            py37h05f1152_2  
pyqt5                     5.12.1                   pypi_0    pypi
pyqt5-sip                 4.19.15                  pypi_0    pypi

$ conda list | grep urllib
urllib3                   1.24.1                   py37_0  

$ conda list | grep certifi
ca-certificates           2018.03.07                    0  
certifi                   2018.11.29               py37_0  

$ lsb_release -a
Distributor ID: Ubuntu
Description:    Ubuntu 18.04.2 LTS
Release:    18.04
Codename:   bionic

```

Note: running xfce4 desktop. This problem occurred also with urllib.

**Comment by haikuginger** [comment]
Before I attempt to replicate, just on a hunch, can you attempt to replicate with a connection to a non-SSL site, such as `neverssl.com`?

Please also go through your issue filing to make sure relevant areas are code-blocked and indented correctly using the triple-backtick format Github supports.

**Comment by mjb-v9-5-2** [comment]
(Ah I see - the code hasn't displayed as code. Thanks for pointing it out. I thought I had blocked the code out using github's own block-the-code-out button. So what can you do? (Well, perhaps check afterwards).

Re. SSL. The same fault occurs when using plain old urllib in the same way, without any certificate. 

I.e. (but with the requests executed inside a pyqt5 event loop):

```
import urllib
u = "https://www.w3.org/2019/03/pressrelease-webauthn-rec.html.en"
l = urllib.request.urlopen(u)
w = l.read().decode('utf-8')

```

**Comment by mjb-v9-5-2** [comment]
I tidied my crappy test code up as well, and removed the scruffy attachments. you don't need to see my working out._.

**Comment by haikuginger** [comment]
Quick update: I'm setting up a Xubuntu VM to debug this. I do suspect that you'll need to eventually go to the pyqt5 people for assistance; since this is happening with the stdlib `urllib` as well, it sounds like some underlying system resources may be getting patched when a pyqt5 application gets initialized. I'll do my best to isolate what specifically might be getting manipulated though.

**Comment by haikuginger** [comment]
Repro build steps:
1. Install Xubuntu 18.04.2
2. `sudo apt install build-essential`
3. `sudo apt install curl`
4. `sudo apt install git`
5. `curl https://pyenv.run | bash`
6. Install pyenv shims in `~/.bashrc`
7. `sudo apt install zlib1g-dev libffi-dev libssl-dev libsqlite3-dev libbz2-dev libreadline-dev`
8. `pyenv install 3.7.3`
9. `pyenv global 3.7.3`
10. `pip install pyqt5==5.12.1 pyqt5-sip==4.19.15 urllib3==1.24.1 certifi==2018.11.29`
11. `sudo apt install qtcreator qt5-default`
12. Create script with above contents (plus an `import sys` line)
13. Run script

I was not able to reproduce the issue using this methodology; the request appeared to complete successfully when not running the pretest. Slight issue was that I had to SIGKILL `python3` from another terminal because there was not a visible window I could close to terminate the `QTApplication` normally.

Closing this issue as it appears to be out-of-scope for `urllib3`.

**Comment by mjb-v9-5-2** [comment]
Well I am glad at least that I might have come near to the possibility of being some use.

FYI, pyqt5 was installed with pip in a conda base environment. I don't know if that makes any difference to your reckonings. In case it does:

```
$ pip show pyqt5

Name: PyQt5
Version: 5.12.1
Summary: Python bindings for the Qt cross platform UI and application toolkit
Home-page: https://www.riverbankcomputing.com/software/pyqt/
Author: Riverbank Computing Limited
Author-email: info@riverbankcomputing.com
License: GPL v3
Location: /home/germentrude/anaconda3/lib/python3.7/site-packages
Requires: PyQt5-sip
Required-by: 

$ pip show pyqt5-sip

Name: PyQt5-sip
Version: 4.19.15
Summary: Python extension module support for PyQt5
Home-page: https://www.riverbankcomputing.com/software/sip/
Author: Riverbank Computing Limited
Author-email: info@riverbankcomputing.com
License: None
Location: /home/germentrude/anaconda3/lib/python3.7/site-packages
Requires: 
Required-by: 

$ conda list | grep pyqt
pyqt                      5.9.2            py37h05f1152_2  
pyqt5                     5.12.1                   pypi_0    pypi
pyqt5-sip                 4.19.15                  pypi_0    pypi
```
